src/Interpreters/executeQuery.cpp

```diff
@@ -563,6 +563,46 @@ void logQueryFinish(
         query_span->addAttributeIfNotZero("clickhouse.written_rows", elem.written_rows);
         query_span->addAttributeIfNotZero("clickhouse.written_bytes", elem.written_bytes);
         query_span->addAttributeIfNotZero("clickhouse.memory_usage", elem.memory_usage);
+
+        if (context)
+        {
+            std::string user_name = context->getUserName();
+            query_span->addAttribute("clickhouse.user", user_name);
+        }
+
+        if (settings[Setting::log_query_settings])
```

Comment:
If this feature is controlled by this setting, it's better to update docs and the setting description in the code to reflect this change.

Reply:
Updated documentation and setting descriptions